atoma-state/src/state_manager.rs

```diff
@@ -0,0 +1,1320 @@
+        .bind(task.deprecated_at_epoch)
+        .bind(task.optimizations)
+        .bind(task.security_level)
+        .bind(task.task_metrics_compute_unit)
+        .bind(task.task_metrics_time_unit)
+        .bind(task.task_metrics_value)
+        .bind(task.minimum_reputation_score)
+        .execute(&self.db)
+        .await?;
+        Ok(())
+    }
+
+    /// Deprecates a task in the database based on its small ID.
+    ///
+    /// This method updates the `is_deprecated` field of a task to `TRUE` in the `tasks` table
+    /// using the provided `task_small_id`.
+    ///
+    /// # Arguments
+    ///
+    /// * `task_small_id` - The unique small identifier for the task to be deprecated.
+    ///
+    /// # Returns
+    ///
+    /// - `Result<()>`: A result indicating success (Ok(())) or failure (Err(StateManagerError)).
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The database query fails to execute.
+    /// - The task with the specified `task_small_id` doesn't exist.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use your_crate::StateManager;
+    ///
+    /// async fn deprecate_task(state_manager: &StateManager, task_small_id: i64) -> Result<(), StateManagerError> {
+    ///     state_manager.deprecate_task(task_small_id).await
+    /// }
+    /// ```
+    #[tracing::instrument(
+        level = "trace",
+        skip_all,
+        fields(task_small_id = %task_small_id)
+    )]
+    pub async fn deprecate_task(&self, task_small_id: i64) -> Result<()> {
+        sqlx::query("UPDATE tasks SET is_deprecated = TRUE WHERE task_small_id = ?")
+            .bind(task_small_id)
+            .execute(&self.db)
+            .await?;
+        Ok(())
+    }
+
+    /// Removes a task from the database based on its small ID.
+    ///
+    /// This method deletes a task from the `tasks` table using the provided `task_small_id`.
+    ///
+    /// # Arguments
+    ///
+    /// * `task_small_id` - The unique small identifier for the task to be removed.
+    ///
+    /// # Returns
+    ///
+    /// - `Result<()>`: A result indicating success (Ok(())) or failure (Err(StateManagerError)).
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The database query fails to execute.
+    /// - The task with the specified `task_small_id` doesn't exist.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// use your_crate::StateManager;
+    ///
+    /// async fn delete_task(state_manager: &StateManager, task_small_id: i64) -> Result<(), StateManagerError> {
+    ///     state_manager.remove_task(task_small_id).await
+    /// }
+    /// ```
+    #[tracing::instrument(
+        level = "trace",
+        skip_all,
+        fields(task_small_id = %task_small_id)
+    )]
+    pub async fn remove_task(&self, task_small_id: i64) -> Result<()> {
+        sqlx::query("DELETE FROM tasks WHERE task_small_id = ?")
```

Comment:
does sqlx cascade deletion? because these is foreign key from another table. That question applies to all deletions.

Reply:
I removed the removals for now. I think it is good for node operators to have access to their full transactional history. It is also possible to remove manually these tables if they grow up and everything is settled.